### PR TITLE
パスワードリセット機能: ドキュメント更新

### PR DESCRIPTION
**Pull request by totodo713**
## 概要
Issue #6 — パスワードリセット機能の実装完了に伴い、未更新だったドキュメントを全面的に更新。

### 新規ドキュメント (3ファイル)
- `docs/api/authentication.md` — 全6認証エンドポイントのAPIリファレンス（リクエスト/レスポンススキーマ、エラーコード、curl例）
- `docs/architecture/password-reset-flow.md` — パスワードリセットのアーキテクチャ文書（Mermaidシーケンス図・状態遷移図、DBスキーマ、セキュリティ設計、コンポーネントマップ）
- `backend/src/main/resources/db/migration/README.md` — マイグレーション索引表 (V1-V12 + R__)

### 既存ドキュメント更新 (5ファイル)
- `README.md` — 認証エンドポイント表追加、Phase 4ステータス追加、ドキュメントリンク追加、マイグレーション数 V1-V7→V1-V12 修正、Basic Auth→Form-based Auth 修正
- `QUICKSTART.md` — MailHogセットアップ手順、パスワードリセットテスト用curl例5ステップ、メール環境変数表を追加
- `docs/user-manual.md` — 「Forgot Password」セクションをGetting Startedに追加
- `backend/src/main/resources/application.yaml` — `frontend.base-url`と`rate-limit`のコメント補足
- `specs/004-user-login-auth/tasks.md` — Phase 5 (US2) のタスク T086-T106 (21件) を完了に更新

### レビュー指摘対応
- `docs/architecture/password-reset-flow.md` のテストファイルパスを実際のコードベースに合わせて修正 (1715ab3)
- `specs/004-user-login-auth/tasks.md` の T092-T094, T101 ファイルパスを `.kt`→`.java` に修正 (26e3b02)
- `specs/004-user-login-auth/tasks.md` の T106 APIクライアントパスを `frontend/app/services/api.ts` に修正 (26e3b02)
- `docs/api/authentication.md` の SecurityConfig パスを `src/main/java/` に修正 (26e3b02)
- `docs/api/authentication.md` の `/logout` Auth Required を `Yes` に修正 (26e3b02)

## 検証チェックリスト
- [ ] 全Markdownファイルが正しくレンダリングされること（テーブル、Mermaid図、リンク）
- [ ] `docs/api/authentication.md` のcurl例が実際のAPIと一致すること
- [ ] README内のドキュメントリンクが正しいファイルに解決すること
- [ ] `application.yaml` の変更がビルドに影響しないこと

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/architecture/password-reset-flow.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+| Auth API Client | `app/services/api.ts` | HTTP calls to backend |
+
+### Tests
+
+| Test | File | Coverage |
+|------|------|----------|
+| PasswordResetService unit | `test/kotlin/.../PasswordResetServiceTest.kt` | Token generation, validation, edge cases |
+| Password reset integration | `test/kotlin/.../PasswordResetIntegrationTest.kt` | Full flow with real DB (Testcontainers) |
+| CSRF protection | `test/kotlin/.../CsrfProtectionTest.kt` | CSRF token enforcement |
+| E2E tests | `test/kotlin/.../PasswordResetE2ETest.kt` | End-to-end API flow |
```
[the remark this answers is not in the document]
Comment (totodo713) [reply docs/architecture/password-reset-flow.md]: Fixed in 1715ab3. Updated the Tests table with actual file paths:
- `PasswordResetIntegrationTest.kt` → `backend/src/test/kotlin/com/worklog/e2e/PasswordResetE2ETest.kt`
- `CsrfProtectionTest.kt` → `backend/src/test/kotlin/com/worklog/api/PasswordResetCsrfTest.kt`
- Added missing entries: `PasswordResetTokenTest`, `PasswordResetTokenRepositoryTest`, `PasswordResetRateLimitIT`

File: specs/004-user-login-auth/tasks.md [hunk 2]
```diff
@@ -189,33 +189,33 @@
...
-- [ ] T096 [US2] Implement token expiration (24 hours) in PasswordResetService.requestReset
-- [ ] T097 [US2] Implement invalidation of previous tokens (set used_at) in PasswordResetService.requestReset
-- [ ] T098 [US2] Add sendPasswordResetEmail method to EmailService with reset link containing token
-- [ ] T099 [US2] Implement token validation (unused, not expired) in PasswordResetService.confirmReset
-- [ ] T100 [US2] Implement password update and session invalidation in PasswordResetService.confirmReset (delete all user sessions and persistent logins)
-- [ ] T101 [US2] Create PasswordResetTokenRepository in `backend/src/main/kotlin/com/worklog/infrastructure/persistence/PasswordResetTokenRepository.kt`
-- [ ] T102 [US2] Add POST /auth/password-reset/request endpoint to AuthController (always returns 200 to prevent email enumeration)
-- [ ] T103 [US2] Add POST /auth/password-reset/confirm endpoint to AuthController with token validation
+- [X] T092 [P] [US2] Create PasswordResetRequestCommand in `backend/src/main/kotlin/com/worklog/application/password/PasswordResetRequestCommand.kt` (email)
```
Comment (totodo713) [inline specs/004-user-login-auth/tasks.md]: **bug**: tasks.md 内の T092〜T094, T101 が `.kt` (Kotlin) パスを参照していますが、実際のファイルはすべて `.java` で `src/main/java/` に配置されています。

例:
- T092: `PasswordResetRequestCommand.kt` → 実際は `PasswordResetRequestCommand.java`
- T101: `PasswordResetTokenRepository.kt` → 実際は `PasswordResetTokenRepository.java`

完了マーク付与に合わせて実態に合わせたパスに修正してください。

File: specs/004-user-login-auth/tasks.md [hunk 3]
```diff
@@ -189,33 +189,33 @@
...
+- [X] T103 [US2] Add POST /auth/password-reset/confirm endpoint to AuthController with token validation
 
 ### Frontend Implementation (US2)
 
-- [ ] T104 [P] [US2] Create password reset request page in `frontend/app/(auth)/password-reset/request/page.tsx` with email form
-- [ ] T105 [P] [US2] Create password reset confirm page in `frontend/app/(auth)/password-reset/confirm/page.tsx` accepting token query param and new password form
-- [ ] T106 [US2] Add API client functions in `frontend/app/lib/api/auth.ts` (requestPasswordReset, confirmPasswordReset)
+- [X] T104 [P] [US2] Create password reset request page in `frontend/app/(auth)/password-reset/request/page.tsx` with email form
+- [X] T105 [P] [US2] Create password reset confirm page in `frontend/app/(auth)/password-reset/confirm/page.tsx` accepting token query param and new password form
+- [X] T106 [US2] Add API client functions in `frontend/app/lib/api/auth.ts` (requestPasswordReset, confirmPasswordReset)
```
Comment (totodo713) [inline specs/004-user-login-auth/tasks.md]: **bug**: T106 で `frontend/app/lib/api/auth.ts` を参照していますが、このファイルは存在しません。
実際の API 関数（`requestPasswordReset`, `confirmPasswordReset`）は `frontend/app/services/api.ts` に実装されています。

File: docs/api/authentication.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+## Source Files
+
+| Component | File |
+|-----------|------|
+| Controller | `backend/src/main/java/com/worklog/api/AuthController.java` |
+| Auth Service | `backend/src/main/java/com/worklog/application/auth/AuthService.java` |
+| Password Reset Service | `backend/src/main/java/com/worklog/application/password/PasswordResetService.java` |
+| Password Validator | `backend/src/main/java/com/worklog/application/validation/PasswordValidator.java` |
+| Email Service | `backend/src/main/java/com/worklog/infrastructure/email/EmailServiceImpl.java` |
+| Security Config | `backend/src/main/kotlin/com/worklog/infrastructure/config/SecurityConfig.kt` |
```
Comment (totodo713) [inline docs/api/authentication.md]: **bug**: SecurityConfig のパスが `src/main/kotlin/` になっていますが、実際のファイルは `backend/src/main/java/com/worklog/infrastructure/config/SecurityConfig.kt` です（`java` ディレクトリ内に `.kt` ファイルが配置されている）。

```suggestion
| Security Config | `backend/src/main/java/com/worklog/infrastructure/config/SecurityConfig.kt` |
```

**Review by totodo713 — COMMENTED** [review]
ドキュメント全体の品質は非常に高く、APIリファレンス・Mermaid図・セキュリティ設計の文書化はよくまとまっています。

ファイルパスの不整合が数箇所あるので修正をお願いします。詳細はインラインコメントを参照してください。

File: docs/api/authentication.md [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+
+---
+
+## Endpoints Overview
+
+| Method | Endpoint | Auth Required | Description |
+|--------|----------|:---:|-------------|
+| POST | `/signup` | No | Register new account |
+| POST | `/login` | No | Authenticate user |
+| POST | `/logout` | No | Invalidate session |
```
Comment (totodo713) [inline docs/api/authentication.md]: **suggestion**: `/logout` の Auth Required が `No` になっていますが、セッション無効化はログイン済みユーザーの操作なので `Yes` が正しいのでは？実装を確認の上修正を検討してください。

File: specs/004-user-login-auth/tasks.md (hunk 2 above)
Comment (totodo713) [reply specs/004-user-login-auth/tasks.md]: Fixed in 26e3b02. T092\~T094, T101 のパスを `.kt`/`kotlin` → `.java`/`java` に修正しました。
